### PR TITLE
`randint` accept ints for 'size'

### PR DESCRIPTION
**Pull request by mtar**
## Description

Issue/s resolved: #913 

## Changes proposed:
- allow ints for size in randint

## Type of change

- Bug fix

## Memory requirements

n/a

## Performance

n/a

## Due Diligence
- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/random.py [hunk 1]
```diff
@@ -528,9 +528,13 @@ def randint(
...
-    shape = tuple(int(ele) for ele in size)
-    if not all(ele > 0 for ele in shape):
-        raise ValueError("negative dimensions are not allowed")
+    try:
+        shape = tuple(int(ele) for ele in size)
+    except TypeError:
+        shape = (int(size),)
+    else:
+        if not all(ele > 0 for ele in shape):
+            raise ValueError("negative dimensions are not allowed")
```
Comment (ClaudiaComito) [inline heat/core/random.py:538]: I was thinking of addressing this like we do for `shape` in `ht.reshape()`:
´´´python
def reshape(a: DNDarray, *shape: Union[int, Tuple[int, ...]], **kwargs) -> DNDarray:
   ...
´´´

Default `size` should be `0`, not `None`. Here a numpy example:
```bash
>>> d = np.random.randint(0,50)
>>> d
1
>>> d = np.random.randint(0,50, 1)
>>> d
array([0])
>>> e = ht.random.randint(0,50)
>>> e
DNDarray([9], dtype=ht.int32, device=cpu:0, split=None)  # ndim=1 but should be 0!
```

File: heat/core/tests/test_random.py [hunk 2]
```diff
@@ -240,6 +240,10 @@ def test_randint(self):
         b = ht.zeros((10,), dtype=ht.int64, split=0)
         self.assertTrue(ht.equal(a, b))
 
+        # size parameter allows int arguments
+        a = ht.random.randint(1, size=10, split=0, dtype=ht.int64)
+        self.assertTrue(ht.equal(a, b))
+
```
Comment (ClaudiaComito) [inline heat/core/tests/test_random.py:246]: Add tests where no size is specified and scalar is returned?

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Thanks a lot @mtar for looking into this right away! I found another inconsistency vs. numpy, see comments.

**Comment by mtar** [comment]
rerun tests

**Comment by mtar** [comment]
reru tests

**Comment by mtar** [comment]
run tests

**Pull request merged**